services.go

```diff
@@ -252,7 +255,79 @@ type SlackService struct {
 // GitLab API docs:
 // https://docs.gitlab.com/ce/api/services.html#slack
 type SlackServiceProperties struct {
-	NotifyOnlyBrokenPipelines bool `json:"notify_only_broken_pipelines"`
+	NotifyOnlyBrokenPipelines bool    `json:"notify_only_broken_pipelines"`
+	NotifyOnlyDefaultBranch   bool    `json:"notify_only_default_branch"`
+	Webhook                   *string `json:"webhook,omitempty"`
+	Username                  *string `json:"username,omitempty"`
+	PushChannel               *string `json:"push_channel,omitempty"`
+	IssueChannel              *string `json:"issue_channel,omitempty"`
+	ConfidentialIssueChannel  *string `json:"confidential_issue_channel,omitempty"`
+	MergeRequestChannel       *string `json:"merge_request_channel,omitempty"`
+	NoteChannel               *string `json:"note_channel,omitempty"`
+	ConfidentialNoteChannel   *string `json:"confidential_note_channel,omitempty"`
+	TagPushChannel            *string `json:"tag_push_channel,omitempty"`
+	PipelineChannel           *string `json:"pipeline_channel,omitempty"`
+	WikiPageChannel           *string `json:"wiki_page_channel,omitempty"`
+}
+
+// UnmarshalJSON implements json.Unmarshaler.
+func (p *SlackServiceProperties) UnmarshalJSON(b []byte) error {
```

Comment:
I would prefer to make a custom `bool` type that has a custom `UnmarshalJSON` method. That way we can reuse the logic (if needed) elsewhere and we don't need this large tailored method in this file.

Reply:
Done. Added a new custom type `BoolValue`.